src/modules/modals/Import.vue

```diff
@@ -240,12 +257,119 @@ export default {
+				} else {
+					showError(t('tables', 'Could not import data due to unknown errors.'))
+					console.debug('error while importing', res)
+				}
+			} catch (e) {
+				console.error(e)
+				return false
+			}
 		},
-		async import() {
+		async previewImportFromUploadFile() {
 			this.loading = true
 			try {
-				const res = await axios.post(generateUrl('/apps/tables/import/' + (this.isElementView ? 'view' : 'table') + '/' + this.element.id), { path: this.path, createMissingColumns: this.getCreateMissingColumns })
+				const url = generateUrl('/apps/tables/importupload-preview/' + (this.isElementView ? 'view' : 'table') + '/' + this.element.id)
+				const formData = new FormData()
+				formData.append('uploadfile', this.selectedUploadFile)
+
+				const res = await axios.post(url, formData, {
+					headers: {
+						'Content-Type': 'multipart/form-data',
+					},
+				})
+
+				if (res.status === 200) {
+					this.preview = res.data
+					this.loading = false
+				} else if (res.status === 401) {
+					console.debug('error while importing', res)
+					showError(t('tables', 'Could not import, not authorized. Are you logged in?'))
+				} else if (res.status === 403) {
+					console.debug('error while importing', res)
+					showError(t('tables', 'Could not import, missing needed permission.'))
+				} else if (res.status === 404) {
+					console.debug('error while importing', res)
+					showError(t('tables', 'Could not import, needed resources were not found.'))
+				} else {
+					showError(t('tables', 'Could not import data due to unknown errors.'))
+					console.debug('error while importing', res)
+				}
+			} catch (e) {
+				console.error(e)
+				return false
+			}
+		},
+		actionImport() {
+			if (this.selectedUploadFile && this.selectedUploadFile.type !== '' && !this.mimeTypes.includes(this.selectedUploadFile.type)) {
+				showWarning(t('tables', 'The selected file is not supported.'))
+				return null
+			}
+
+			if (!this.validateColumnsConfig()) {
+				return null
+			}
+
+			if (this.selectedUploadFile) {
+				this.importFromUploadFile()
+				return null
+			}
+
+			if (this.path === '') {
+				showWarning(t('tables', 'Please select a file.'))
+				this.pathError = true
+				return null
+			}
+			this.pathError = false
+			this.importFromPath()
+		},
+		validateColumnsConfig() {
+			const existColumnCount = {}
+			for (const column of this.columnsConfig) {
+				if (column.action === 'exist') {
+					if (!column.existColumn) {
+						showWarning(t('tables', 'Please select column for mapping.'))
+						return false
+					}
+					if (existColumnCount[column.existColumn.id] === undefined) {
+						existColumnCount[column.existColumn.id] = 1
+					} else {
+						existColumnCount[column.existColumn.id]++
+					}
+				}
+			}
+			if (Object.values(existColumnCount).some(count => count > 1)) {
+				showWarning(t('tables', 'Cannot map same exist column for multiple columns.'))
+				return false
+			}
```

Comment:
Will update it. The selected column will be disabled in dropdown. I will still keep the warning just in case a column is selected multiple times some how.

Reply:
Sounds good :)